src/Relation/Binary/Construction/Free/Supremum/StrictOrder.agda

```diff
@@ -0,0 +1,112 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Consequences on strict orders of freely adding an infimum to a Set
+------------------------------------------------------------------------
+
+open import Relation.Binary
+
+module Relation.Binary.Construction.Free.Supremum.StrictOrder
+       {a r} {A : Set a} (_<_ : Rel A r) where
+
+open import Level
+open import Data.Product
+open import Function
+open import Function.Equivalence using (equivalence)
+open import Relation.Nullary
+import Relation.Nullary.Decidable as Dec
+import Relation.Binary.PropositionalEquality as P
+open import Relation.Binary.Construction.Free.Supremum
+
+data _<⁺_ : Rel (A ⁺) r where
+  []<[]  : {k l : A} → k < l → [ k ] <⁺ [ l ]
+  [_]<⊤⁺ : (k : A)           → [ k ] <⁺ ⊤⁺
+
+[]<[]⁻¹ : ∀ {k l} → [ k ] <⁺ [ l ] → k < l
+[]<[]⁻¹ ([]<[] p) = p
+
+<⁺-asym : Asymmetric _<_ → Asymmetric _<⁺_
+<⁺-asym <-asym ([]<[] p) ([]<[] q) = <-asym p q
+<⁺-asym <-asym [ k ]<⊤⁺  ()
+
+<⁺-trans : Transitive _<_ → Transitive _<⁺_
+<⁺-trans <-trans ([]<[] p) ([]<[] q) = []<[] (<-trans p q)
+<⁺-trans <-trans ([]<[] p) [ k ]<⊤⁺  = [ _ ]<⊤⁺
+
+<⁺-dec : Decidable _<_ → Decidable _<⁺_
+<⁺-dec <-dec [ k ] [ l ] = Dec.map (equivalence []<[] []<[]⁻¹) (<-dec k l)
+<⁺-dec <-dec [ k ] ⊤⁺    = yes [ k ]<⊤⁺
+<⁺-dec <-dec ⊤⁺    [ l ] = no (λ ())
+<⁺-dec <-dec ⊤⁺    ⊤⁺    = no (λ ())
+
+<⁺-irrelevance : Irrelevant _<_ → Irrelevant _<⁺_
+<⁺-irrelevance <-irrelevance ([]<[] p) ([]<[] q) = P.cong _ (<-irrelevance p q)
+<⁺-irrelevance <-irrelevance [ k ]<⊤⁺  [ .k ]<⊤⁺ = P.refl
+
+module _ {e} {_≈_ : Rel A e} where
+
+  open import Relation.Binary.Construction.Free.Supremum.Pointwise _≈_
+
+  <⁺-tri : Trichotomous _≈_ _<_ → Trichotomous _≈⁺_ _<⁺_
+  <⁺-tri <-tri [ k ] [ l ] with <-tri k l
+  <⁺-tri <-tri [ k ] [ l ] | tri< a ¬b ¬c = tri< ([]<[] a) (¬b ∘ []≈[]⁻¹) (¬c ∘ []<[]⁻¹)
```

Comment:
The start of these can be replaced with `...` to shorten the length of the lines.